protocol-designer/src/utils/index.ts

```diff
@@ -189,21 +189,14 @@ export const getStagingAreaAddressableAreas = (
 export function getMatchingTipLiquidSpecsFromSpec(
   pipetteSpecs: PipetteV2Specs,
   volume: number,
-  tiprackUri: string
+  tiprackDef: LabwareDefinition2
```

Comment:
If I'm understanding this, I think this type should be a union including null
```suggestion
  tiprackDef: LabwareDefinition2 | null
```

Reply:
So the caller **should** always supply a `tiprackDef`, so I'd like to leave this declaration as not-null.

But in practice, like the comment says, the implementation in the caller of this function is wrong, and will fail to find the `tiprackDef` for custom tipracks, but that is something we will fix in a future PR.